client/txn_test.go

```diff
@@ -288,7 +288,7 @@ func TestEndWriteRestartReadOnlyTransaction(t *testing.T) {
 			t.Errorf("expected error: %t, got error: %v", !success, err)
 		}
 		if !reflect.DeepEqual(expCalls, calls) {
-			t.Fatalf("expected %v, got %v", expCalls, calls)
+			t.Errorf("expected %v, got %v", expCalls, calls)
```

Comment:
Continuing the test when it has already failed isn't necessarily useful. I'd revert this and the other changes below.